conda_lock/conda_lock.py

```diff
@@ -165,21 +165,27 @@ def print_proc(proc):
     try:
         proc.check_returncode()
     except subprocess.CalledProcessError:
-        try:
-            err_json = json.loads(proc.stdout)
-            message = err_json["message"]
-        except json.JSONDecodeError as e:
-            print(f"Failed to parse json, {e}")
-            message = ""
-
         print(f"Could not perform conda install using {file} lock file into {prefix}")
-        if message:
-            print(message)
+        print(_handle_subprocess_stdout(proc.stdout))
         print_proc(proc)
-
         sys.exit(1)
 
 
+def _handle_subprocess_stdout(stdout, retry=True):
+    try:
+        err_json = json.loads(stdout)
+        if err_json.get("exception_name") == "CondaMultiError":
+            return "\n\n".join(error["message"] for error in err_json["errors"])
+        return err_json["message"]
```

Comment:
Assumption:

When an exception is a `CondaMultiError` it contains an `errors` property that contains a list of error objects (and each error contains a `message` property). Otherwise, the error contains a `message` property on root level.

Unfortunately I haven't seen an error yet where `exception_name  != "CondaMultiError"`, so this is more of a guess. If you have more examples for errors, that would be really useful here.